apps/extension/src/core/utils/utils.ts

```diff
@@ -66,3 +66,15 @@ export function formatDate(timestampMs: number): string {
 export function assertNever(value: never): never {
   throw new Error(`Unexpected value: ${value}`)
 }
+
+export function extractExtensionFromURL(url: string): string {
+  let extension: string = ""
+  const urlObject = new URL(url)
+  const pathComponents = urlObject.pathname.split(".")
+  const extensionFromURL = pathComponents.pop()
```

Comment:
better to use functional syntax: `const extensionFromURL = pathComponents[pathComponents.length - 1]` to avoid mutating pathComponents, which could surprise developers who use variables like it somewhere else